openpype/hosts/maya/plugins/load/load_redshift_proxy.py

```diff
@@ -137,6 +137,16 @@ def create_rs_proxy(self, name, path):
         cmds.connectAttr("{}.outMesh".format(rs_mesh),
                          "{}.inMesh".format(mesh_shape))
 
+        # TODO: use the assigned shading group as shaders if existed
+        # assign default shader to redshift proxy
+        shader_grp = next(
+            (
+                shader_group for shader_group in
+                cmds.ls("initialShadingGroup", type="shadingEngine")
+            )
+        )
+        cmds.sets(mesh_shape, forceElement=shader_grp)
```

Comment:
There really is no need to use `next()` since you're querying explicitly by name. Also, your `next` check does not allow the shading engine to not exist since it returns no default value - as such, you're basically in your code now already assuming a shading engine exists with name `"initialShadingGroup"`. Hence you can just as well remove the check for existence altogether and just do `cmds.sets(mesh_shape, forceElement="initialShadingGroup")`

```suggestion
        if cmds.ls("initialShadingGroup", type="shadingEngine"):
            cmds.sets(mesh_shape, forceElement=shader_grp)
```

Or if you're sure it exists anyway you can also remove the check itself and just assign `"initialShadingGroup"` directly without the check.

```suggestion
        cmds.sets(mesh_shape, forceElement="initialShadingGroup")
```

Reply:
gotcha. I was doing this as there might be some possible ways to get the attributes from proxy materials and applied the shader directly to the proxy materials. But we can always do this later.

Reply:
> as there might be some possible ways to get the attributes from proxy materials and applied the shader directly to the proxy materials.

Sorry, could you elaborate? What do you mean?

Reply:
> > as there might be some possible ways to get the attributes from proxy materials and applied the shader directly to the proxy materials.
> 
> Sorry, could you elaborate? What do you mean?
I mean the shader group shown in the slot of proxy materials in redshift proxy mesh.
![image](https://github.com/ynput/OpenPype/assets/64118225/a5bc2b5a-f523-4169-93ef-283048572051)

Reply:
Not sure if I'd expect that to happen on load. Decent feature though, just not sure whether that logic should be here.

Reply:
> Not sure if I'd expect that to happen on load. Decent feature though, just not sure whether that logic should be here.

@LiborBatek can you give some advice on this?